src/org/apache/storm/metrics2/RocksConnector.java

```diff
@@ -97,9 +97,12 @@ public void remove()
             String key = new String(iterator.key());
             String[] elements = key.split("\\|");
 
-            if (!(settings.containsKey("metric") && !elements[0].equals(settings.get("metric")) ||
-                    settings.containsKey("compId") && !elements[0].equals(settings.get("compId")) ||
-                    settings.containsKey("topoId") && !elements[3].equals(settings.get("topoId")))){
+            if (!(settings.containsKey("timeStart") && !(Integer.parseInt(elements[0]) >= Integer.parseInt(settings.get("timeStart").toString())) ||
+                    settings.containsKey("timeEnd") && !(Integer.parseInt(elements[0]) <= Integer.parseInt(settings.get("timeEnd").toString())) ||
+                    settings.containsKey("metric") && !elements[1].equals(settings.get("metric")) ||
+                    settings.containsKey("compId") && !elements[2].equals(settings.get("compId")) ||
+                    settings.containsKey("topoId") && !elements[3].equals(settings.get("topoId"))
```

Comment:
Why not just iterate over settings here rather than doing this boolean logic? I think it would be simpler to iterate over every key/value in the map, and then check if the element from rocks matches each of the settings in the maps